### ISSUE TITLE
Creating a MACRO in pg_catalog or information_schema causes duckdb to crash

### ISSUE THREAD

**Issue by LasseJacobs** [duckdb/duckdb#3690]
#### What happens?
Creating a MACRO in pg_catalog or information_schema causes duckdb to crash. 

#### To Reproduce
- open duckdb
- run:
```sql
create macro information_schema.foo(a) as a;
--or
create macro pg_catalog.foo(a) as a;
```
- crash! "52896 segmentation fault  ./duckdb"

#### Environment (please complete the following information):
 - OS: MacOS Big Sur (M1)
 - DuckDB Version: I am using a self-compiled version (compiled a few days ago from master). Running `--version` gives me this: v0.3.5-dev602 ee6a6986d 
 - DuckDB Client: cli

#### Before Submitting

- [y] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source. -> I don't have access to a linux machine right now but I will try to use one of the official binaries and test on linux later (there are no binaries for macos correct?).

- [y] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

#### Some context
I am trying to create this function: `pg_catalog.pg_get_userbyid`. I am creating a proxy that speaks the postgres wire protocol so I can interact with duckdb remotely using standard postgres tools. Running queries works fine but it will fail when using the `\commands` for example `\d`. Not sure if this is a realistic goal or not but it is just a project for fun.

**Comment by LasseJacobs** [comment]
I did some more testing:
-  `CREATE TABLE` and `CREATE VIEW` also fail when the schema is `pg_catalog`. 
- The issue also occurs on linux (ubuntu 20.04) with the 'official' binaries (v0.3.4)

**Comment by Mytherin** [comment]
Thanks for the report! I can confirm this crashes. I have pushed a fix in #3691

**Comment by LasseJacobs** [comment]
I recompiled and can confirm that it is working now. Thank you for the quick response!
